nearcore/src/lib.rs

```diff
@@ -287,6 +288,18 @@ pub fn start_with_config_and_synchronization(
         hash: *genesis_block.header().hash(),
     };
 
+    // State Sync actors
+    let client_adapter_for_sync = Arc::new(LateBoundSender::default());
```

Comment:
not a huge deal, but these are only used if the config is `SyncConfig::Peer`, so could just initialize them only if that is the case. Maybe they could also be `Option`s and then below the `bind()` calls are only necessary in that case

Reply:
Good catch!